components/form/ServiceNameSelect.vue

```diff
@@ -0,0 +1,256 @@
+    },
+
+    searchable: {
+      type:    Boolean,
+      default: true,
+    },
+
+    taggable: {
+      type:    Boolean,
+      default: true,
+    },
+
+    selectLabel: {
+      type:    String,
+      default: '',
+    },
+
+    optionLabel: {
+      type:    String,
+      default: 'label',
+    },
+
+    options: {
+      type:     Array,
+      required: true,
+    },
+
+    selectBeforeText: {
+      type:    Boolean,
+      default: true,
+    },
+
+    reduce: {
+      default: (e) => {
+        if (e && typeof e === 'object' && e.value !== undefined) {
+          return e.value;
+        }
+
+        return e;
+      },
+      type: Function
+    },
+  },
+
+  data() {
+    return { selected: this.value || null };
+  },
+
+  computed: {
+    serviceNameNew() {
+      if (this.value !== null) {
+        const findSelected = this.options.find(option => this.value === option.metadata.name);
+
+        if (!findSelected) {
+          return true;
+        }
+      }
+
+      return false;
+    }
+  },
+
+  methods: {
+    focus() {
+      const comp = this.$refs.text;
+
+      if (comp) {
+        comp.focus();
+      }
+    },
+
+    change() {
+      const selectedOption = this.options.find(option => option.metadata.id === this.selected.serviceAccount);
+
+      if (selectedOption) {
+        this.$emit('input', selectedOption.metadata.name);
+      }
+    },
+
+    blurCreate(e) {
+      const searched = e;
+
+      if (searched) {
+        this.selected = searched;
+        this.$emit('input', searched);
+      }
+    },
+
+    clearSearch(event) {
+      this.selected = null;
+      this.$emit('input', null);
+
+      event.preventDefault();
+    }
+  },
+};
+</script>
+
+<template>
+  <div>
+    <div class="spacer"></div>
+    <div class="row mb-10">
+      <h3 class="col span-6">
+        {{ t('workload.serviceAccountName.label') }}
+      </h3>
+    </div>
+    <div class="row">
+      <div class="col span-5">
+        <div
+          :class="{ 'select-after': !selectBeforeText }"
+          class="servicename-select input-container container-flex"
+          @input="change"
+        >
+          <LabeledSearchSelect
+            v-if="selectLabel"
+            v-model="selected"
+            :label="selectLabel"
+            :class="{ 'in-input': !isView }"
+            :options="options"
+            :searchable="true"
+            :clearable="true"
+            :disabled="disabled || isView"
+            :taggable="taggable"
+            :create-option="(name) => (name)"
+            :reduce="(name) => `${name}`"
+            :multiple="false"
+            :mode="mode"
+            :option-label="optionLabel"
+            :placement="$attrs.placement ? $attrs.placement : null"
+            :v-bind="$attrs"
+            @input="change"
+            @blur-create="blurCreate"
+          />
+        </div>
+      </div>
+      <button class="btn btn-sm clear-btn role-secondary col span-1" @click="clearSearch($event);">
```

Comment:
This shows up on the 'view config' page (so needs to take `mode` into account, and be hidden if is `_VIEW`)